crates/edr_solidity_tests/src/multi_runner.rs

```diff
@@ -414,38 +414,17 @@ impl<
             find_time,
         );
 
-        let handle = tokio::runtime::Handle::current();
+        let tokio_handle = tokio::runtime::Handle::current();
 
-        let this = Arc::new(self);
-        let args = contracts.into_iter().zip(std::iter::repeat((
-            this,
-            fork,
-            filter,
-            on_test_suite_completed_fn,
-        )));
-
-        handle.spawn(async {
-            futures::stream::iter(args)
-                .for_each_concurrent(
-                    Some(num_cpus::get()),
-                    |((id, contract), (this, fork, filter, on_test_suite_completed_fn))| async move {
-                        tokio::task::spawn_blocking(move || {
-                            let handle = tokio::runtime::Handle::current();
-                            let result =
-                                this.run_tests(&id, &contract, fork, filter.as_ref(), &handle);
-
-                            on_test_suite_completed_fn(
-                                SuiteResultAndArtifactId {
-                                    artifact_id: id,
-                                    result,
-                                },
-                            );
-                        })
-                        .await
-                        .expect("failed to join task");
-                    },
-                )
-                .await;
+        contracts.into_par_iter().for_each(|(id, contract)| {
```

Comment:
My understanding is that this blocks until all of the array elements have been processed, whereas `handle.spawn` in the previous version would return control immediately while the task runs asynchronously.

Am I understanding that correctly? Is there a reason why that's not an issue?

This is my only question about the PR.